### PR TITLE
[shader-slang] Updating to 2024.10

### PR DESCRIPTION
**Pull request by Honeybunch**
In updating this port I found that `slangc` when installed as a tool was missing some dependencies so I made sure that they got installed properly. I guess there was no way for `vcpkg_copy_tools` to detect that `slangc` / `slangd` need these other shared libraries to function.

- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [X] SHA512s are updated for each updated download.
- [X] The "supports" clause reflects platforms that may be fixed by this new version.
- [X] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [X] Any patches that are no longer applied are deleted from the port's directory.
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/shader-slang/portfile.cmake [hunk 1]
```diff
@@ -86,8 +86,15 @@ file(GLOB libs
 )
 file(INSTALL ${libs} DESTINATION "${CURRENT_PACKAGES_DIR}/lib")
 
+file(GLOB dyn_libs
+	"${BINDIST_PATH}/lib/*.dylib"
+	"${BINDIST_PATH}/lib/*.so"
+	"${BINDIST_PATH}/bin/*.dll"
```
Comment (dg0yt) [inline ports/shader-slang/portfile.cmake:92]: This seems redundant, given line 97 below.

File: ports/shader-slang/portfile.cmake [hunk 2]
```diff
@@ -36,7 +36,7 @@ if(key STREQUAL "macosx-x64" OR VCPKG_SHADER_SLANG_UPDATE)
 		ARCHIVE
 		URLS "https://github.com/shader-slang/slang/releases/download/v${VERSION}/slang-${VERSION}-macos-x86_64.zip"
 		FILENAME "slang-${VERSION}-macos-x86_64.zip"
-		SHA512 50a710f4432ab1b9cf08e3b490b9e663e7f7a20ed8b8f19fd0d9650cef53e637f66f53a87c1df11b521faaf85fd2e61434f457487cc646debdb240607eaf952f
+		SHA512 5761f44c0fb9c2da5b3f2457bda7b5f1a0f45cc5711f5c2c72df3a5743e6663996b6207834e0f6645ac99a655318b8ce96669b38ebc9f82daace8949010ff20b
```
Comment (dg0yt) [inline ports/shader-slang/portfile.cmake:39]: Can you pleae remove the three lines starting at  `set(unavailable_for_x64 ...)`?
This was a temporary workaround which seems obsolete now.

**Review by dg0yt — COMMENTED** [review]
There are two things needed for `vcpkg_copy_tools`: The runtime lib must be in the function's search path, and the dependency must be announced to the dynamic linker. Modules/plugins loaded later during  runtime cannot be detected.

File: ports/shader-slang/portfile.cmake (hunk 1 above)
Comment (Honeybunch) [reply ports/shader-slang/portfile.cmake:92]: Oops yep

**Review by MonicaLiu0311 — APPROVED** [review]

File: ports/shader-slang/portfile.cmake [hunk 3]
```diff
@@ -98,6 +103,8 @@ if(NOT VCPKG_BUILD_TYPE)
 	endif()
 endif()
 
+# Must manually copy some tool dependencies since vcpkg can't copy them automagically for us
+file(INSTALL ${dyn_libs} DESTINATION "${CURRENT_PACKAGES_DIR}/tools/shader-slang/")
```
Comment (vicroms) [inline ports/shader-slang/portfile.cmake:107]: Have you tested that this makes the executable work on non-Windows?

Comment (Honeybunch) [reply ports/shader-slang/portfile.cmake:107]: Testing now

Comment (Honeybunch) [reply ports/shader-slang/portfile.cmake:107]: Seems to work across my test suite for my project. Looks like osx and linux work fine with slangc https://github.com/HoneybunchBuilder/toybox/pull/4

**Comment by Honeybunch** [comment]
ACK! I think I was wrong. No idea how I avoided getting bit by this until now. There is a problem with this. 
image
Sorry for the screenshot, over VNC. 
Seems that the RPATH of `slangc` can't be updated by vcpkg and as warned it screws with code-signing. I am not enough of an expert in code-signing to know exactly what's up or if there's a straightforward fix. 

If you're struggling with macos failing to use slangc with "Killed 9:" this is why 😢

**Comment by Honeybunch** [comment]
ACK! I think I was wrong. No idea how I avoided getting bit by this until now. There is a problem with this. 
image
Sorry for the screenshot, over VNC. 
Seems that the RPATH of `slangc` can't be updated by vcpkg and as warned it screws with code-signing. I am not enough of an expert in code-signing to know exactly what's up or if there's a straightforward fix. 

If you're struggling with macos failing to use slangc with "Killed 9:" this is why 😢

**Comment by Honeybunch** [comment]
Nevermind, I needed to sleep. I just realized I can turn off rpath modifications for osx and that worked. I should have a patch up soon

**Pull request merged**